web/packages/teleport/src/components/FormLogin/FormIdentifierFirst.tsx

```diff
@@ -0,0 +1,275 @@
+/**
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useCallback, useEffect, useState } from 'react';
+
+import {
+  Box,
+  ButtonLink,
+  ButtonPrimary,
+  Flex,
+  Indicator,
+  Input,
+  Text,
+} from 'design';
+import * as Alerts from 'design/Alert';
+import ButtonSso, { guessProviderType } from 'shared/components/ButtonSso';
+import { useRefAutoFocus } from 'shared/hooks';
+import { Attempt, useAsync } from 'shared/hooks/useAsync';
+import { AuthProvider } from 'shared/services';
+
+import ResourceService from 'teleport/services/resources';
+import { storageService } from 'teleport/services/storageService';
+
+import SSOButtonList from './SsoButtons';
+
+type Props = {
+  onLoginWithSso(provider: AuthProvider): void;
+  /**
+   * onUseLocalLogin is called to switch the view to the local login form.
+   */
+  onUseLocalLogin(): void;
+};
+
+export function FormIdentifierFirst({
+  onLoginWithSso,
+  onUseLocalLogin,
+}: Props) {
+  const [resourceService] = useState(() => new ResourceService());
+
+  const [rememberedUsername, setRememberedUsername] = useState<string>(
+    storageService.getRememberedSsoUsername()
+  );
+  const [username, setUsername] = useState<string>(rememberedUsername);
+  const [connectors, setConnectors] = useState<AuthProvider[]>([]);
+
+  useEffect(() => {
+    if (rememberedUsername) {
+      fetchMatchingConnectors(rememberedUsername);
+    }
+  }, [rememberedUsername]);
+
+  const [fetchAttempt, fetchMatchingConnectors] = useAsync(
+    useCallback(
+      async (username: string) => {
+        const connectors =
+          await resourceService.getUserMatchedAuthConnectors(username);
+        if (connectors.length === 0) {
+          if (rememberedUsername) {
+            // If we have a remembered username but no connectors, we clear the remembered username.
+            storageService.clearRememberedSsoUsername();
+            setRememberedUsername('');
+            setUsername('');
+            return;
+          }
+          throw new Error(`No SSO connectors found for user: ${username}`);
+        }
+        // If there isn't a remembered username, and there is only one matching connector, we take them straight to the IdP.
+        if (connectors.length === 1 && !rememberedUsername) {
```

Comment:
can we have tests for this logic of remembering username based on connector response?